libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -106,18 +106,75 @@ void ExpressionCompiler::appendStateVariableAccessor(VariableDeclaration const&
 		if (auto mappingType = dynamic_cast<MappingType const*>(returnType.get()))
 		{
 			solAssert(CompilerUtils::freeMemoryPointer >= 0x40, "");
-			solUnimplementedAssert(
-				!paramTypes[i]->isDynamicallySized(),
-				"Accessors for mapping with dynamically-sized keys not yet implemented."
-			);
+
 			// pop offset
 			m_context << Instruction::POP;
-			// move storage offset to memory.
-			utils().storeInMemory(32);
-			// move key to memory.
-			utils().copyToStackTop(paramTypes.size() - i, 1);
-			utils().storeInMemory(0);
-			m_context << u256(64) << u256(0) << Instruction::KECCAK256;
+			if (paramTypes[i]->isDynamicallySized())
+			{
+				// stack: <keys..> <slot position>
+
+				// copy key[i] to top.
+				utils().copyToStackTop(paramTypes.size() - i + 1, 1);
+
+				// stack: <keys..> <slot position> <key1>
+				// Load key length to stack
```

Comment:
You can only use the shortcut if the type is actually a `string` or `bytes` type - please see the KECCAK256 function implementation in ExpressionCompiler. I would also advise to use `m_context.appendInlineAssembly(...)` here.

Reply:
But according to the documentation `string` and `bytes` are the only dynamic types allowed for mappings?